app/models/piece.rb

```diff
@@ -124,13 +124,21 @@ def never_moved?
     updated_at == created_at
   end
 
+  def active_player
+    if self.game.move_counter % 2 == 0
+      return self.game.white_user
+    end
+    return self.game.black_user
+  end
+
   def move_to!(x,y)
-    return false unless valid_move?(x,y)
-    capture_piece_at(x,y)
+    return false unless valid_move?(x, y) && self.active_player.id == self.user_id
```

Comment:
Could you clarify this for me?  On my local host, it seems to do what I expect.  It only allows white to move white, black to move black, and only when it is their turn.  Is it going to cause the problems later when we have it all on one page in real-time? Just trying to understand :-/